topology: add equalizer to cml rt5682 and hda generic topology

fredoh9 opened the pull request:
Add EQIIR and EQFIR to first playback pipeline. This help to expand more
equalizer testing across platforms.

Signed-off-by: Fred Oh <fred.oh@linux.intel.com>

Changes files from sof-cml-rt5682-kwd.m4, sof-cml-rt5682.m4, sof-hda-generic.m4 respectively. One line is changed from pipe-volume-playback.m4 to pipe-eq-volume-playback.m4

Tested in CML Hatch with RT5682 and WHL HDA generic. Plan to expand daily EQ testing with CML_HEL_RT5682, CLF_RVP_HDA, ICL_RVP_HDA.

lgirdwood commented on tools/topology/sof-cml-eq-rt5682-kwd.m4:
```diff
@@ -0,0 +1,233 @@
+define(`SSP_MCLK_RATE', ifelse(PLATFORM, `icl', `19200000',
+	ifelse(PLATFORM, `whl', `24000000',
+		ifelse(PLATFORM, `cml', `24000000', `'))))
+
```
I'd prefer to see this taken care of by ```include(PLATFORM.m4)``` where PLATFORM.m4 sets the correct SSP port, name and clock

lgirdwood replied:
@fredoh9 this would be good if you could make the same change to all topologies that have machine specific conditional logic as it would be a great simplification as a separate PR

fredoh9 replied:
@lgirdwood Sure, I will review PLATFORM specific changes.

singalsu commented on tools/topology/sof-cml-eq-rt5682-kwd.m4:
```diff
@@ -0,0 +1,233 @@
+#
+# Define the pipelines
+#
+# PCM0 <---> volume <----> SSP(SSP_INDEX, BE link 0)
```
Please add EQ to this PCM0 pipeline description

singalsu commented on tools/topology/sof-hda-generic-eq.m4:
```diff
@@ -0,0 +1,185 @@
+#
+# Define the pipelines
+#
+# PCM0P --> volume     (pipe 1) --> HDA Analog (HDA Analog playback)
```
Also add EQ here

juimonen commented:
@fredoh9 probably good to separate the changes for different platforms into separate commits.

fredoh9 commented:
@juimonen Sure, I can separate the commit, this for sof-cml-rt5682-kwd.m4 and the other for sof-hda-generic.m4. Other files are not used for now.
I'm working on https://github.com/thesofproject/sof/pull/2461 as a cleanup first.

fredoh9 commented:
@juimonen @singalsu I rebased and made separate commit per topology.

lgirdwood commented on tools/topology/sof-cml-eq-rt1011-rt5682.m4:
```diff
@@ -0,0 +1,70 @@
+#Use BCLK delay in SSP_CONFIG_DATA only on supporting version
+
+ifelse(SOF_ABI_VERSION_3_9_OR_GRT, `1',
+`DAI_CONFIG(SSP, 1, 6, SSP1-Codec,
```
Conditional logic again. The only difference between the two configs is SSP_CONFIG_DATA, can we define SSP_CONFIG_DATA in PLATFORM.m4 (which will have the ABI check)

fredoh9 replied:
@lgirdwood thank you for pin-pointing the problem. I will fix and resubmit the PR.

fredoh9 replied:
Sorry for delay. Still working on a patch. Struggling to fix some differences after my new patch. Trying to finalizing the patch soon.

fredoh9 replied:
I submitted https://github.com/thesofproject/sof/pull/2530. I was debugging some issues, I decided better share my finds and get input from reviewers. If #2530 is revised and approved, I plan to update this also. Now I can see much bigger picture, there is no point to create new file for one line change. I will add makefile variable and put some conditions.

fredoh9 commented:
@juimonen @singalsu @lgirdwood I updated with makefile changes without making new files. I think this is right and better way to keep EQ topology. What do you think? And we don't need to wait CML topology clean up.

I see some build errors in ICL but the error is not relevant.

lgirdwood commented on tools/topology/CMakeLists.txt:
```diff
@@ -105,6 +105,7 @@ set(TPLGS
 	"sof-cml-rt5682-max98357a\;sof-cml-rt5682-max98357a\;-DPLATFORM=cml"
 	"sof-cml-demux-rt5682-max98357a\;sof-cml-demux-rt5682-max98357a\;-DPLATFORM=cml"
 	"sof-cml-rt1011-rt5682\;sof-cml-rt1011-rt5682\;-DPLATFORM=cml"
+	"sof-cml-rt1011-rt5682\;sof-cml-eq-rt1011-rt5682\;-DPLATFORM=cml\;-DEQ=yes"
```
best to use "enabled" instead of "yes"

fredoh9 commented:
@lrgirdwo Only change I made after your approval, changed makefile variable from EQ to PPROC, Playback Processing. This is an effort to be align with https://github.com/thesofproject/sof/pull/2558

@singalsu @xiulipan I updated PPROC variable here also.

lgirdwood commented:
SOFCI TEST

xiulipan commented on tools/topology/CMakeLists.txt:
```diff
@@ -23,9 +23,11 @@ set(TPLGS
+	"sof-hda-generic\;sof-hda-generic\;-DCHANNELS=0\;-DPPROC=volume"
+	"sof-hda-generic\;sof-hda-generic-2ch\;-DCHANNELS=2\;-DPPROC=volume"
+	"sof-hda-generic\;sof-hda-generic-4ch\;-DCHANNELS=4\;-DPPROC=volume"
+        "sof-hda-generic\;sof-hda-generic-eq-2ch\;-DCHANNELS=2\;-DPPROC=eq-volume"
```
intend issue here.

xiulipan commented on tools/topology/CMakeLists.txt:
```diff
@@ -104,7 +106,8 @@ set(TPLGS
 	"sof-cml-demux-rt5682-max98357a\;sof-cml-demux-rt5682-max98357a\;-DPLATFORM=cml"
-	"sof-cml-rt1011-rt5682\;sof-cml-rt1011-rt5682\;-DPLATFORM=cml"
+	"sof-cml-rt1011-rt5682\;sof-cml-rt1011-rt5682\;-DPLATFORM=cml\;-DPPROC=volume"
+	"sof-cml-rt1011-rt5682\;sof-cml-eq-rt1011-rt5682\;-DPLATFORM=cml\;-DPPROC=eq-volume"
```
what would be eq used for? DMIC or rt1011, can we have some style here about where to pu the eq?

fredoh9 replied:
This is adding eq pipeline at first playback pipe, which is rt5682. RT1011 is for speaker and pipeline 7. I chose sof-cml-rt1011-rt5682 because it is used in cml helios.

lgirdwood commented:
Looks like CI is incomplete, lets rerun after server reboot.

lgirdwood commented:
SOFCI TEST

fredoh9 commented:
I checked Internal Intel CI System page, it's false alarm. DEVICE_TEST results look normal as well

lgirdwood commented:
@zrombel @wwittbrx we are only seeing partial results from interal CI (all green) but result is fail. Can you check.

zrombel commented:
@lgirdwood It was a QB issue, since your comment PR was rebuilt and now everything looks fine.

jajanusz commented:
@lgirdwood Travis failed to report success. Admin has to merge it.

lgirdwood commented:
Travis all green.
